muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+import {
+  displayNotification,
+  DisplayNotificationTriggerType,
+} from "~/actions/base/notifications";
+import { connect, Dispatch } from "react-redux";
+import { AnyActionType } from "~/actions";
+import { bindActionCreators } from "redux";
+import { i18nType } from "~/reducers/base/i18n";
+import { StatusType } from "~/reducers/base/status";
+import { WorkspaceMaterialReferenceType } from "~/reducers/workspaces";
+import { StateType } from "~/reducers";
+import { useCourseCarousel } from "~/components/general/carousel/hooks/use-course-carousel";
+import WorkspaceSignup from "~/components/coursepicker/dialogs/workspace-signup";
+import Button from "~/components/general/button";
+import "~/sass/elements/panel.scss";
+import "~/sass/elements/item-list.scss";
+
+/**
+ * NextStudiesPanelProps
+ */
+interface NextStudiesPanelProps {
+  i18n: i18nType;
+  status: StatusType;
+  lastWorkspace: WorkspaceMaterialReferenceType;
+  displayNotification: DisplayNotificationTriggerType;
+}
+
+/**
+ * NextStudiesPanel
+ * @param props NextStudiesPanelProps
+ * @returns JSX.Element
+ */
+const NextStudiesPanel: React.FC<NextStudiesPanelProps> = (props) => {
+  const { courseCarousel } = useCourseCarousel(
+    props.status.userSchoolDataIdentifier,
+    props.status.userId,
+    props.status.isStudent,
+    props.displayNotification
+  );
+
```

Comment:
You are passing more parameters than function can hold. There is no boolean parameter to check if user is student. If you have changed useCourseCarousel custom hook, then changes are not shown correctly